lib/elixir/lib/enum.ex

```diff
@@ -2584,8 +2584,9 @@ defmodule Enum do
   Enumerates the `enumerable`, by removing the elements for which
   function `fun` returned duplicate items.
 
-  The function `fun` maps every element to a term which is used to
-  determine if two elements are duplicates.
+  The function `fun` maps every element to a term and then compares the
```

Comment:
That is better, and I agree about the ambiguity of what `fun` is doing in my proposal. How about:

The function `fun` maps every element to a term. `uniq_by/2` compares the equality of these terms to determine which elements in the `enumerable` are duplicates.